python/taichi/lang/impl.py

```diff
@@ -16,12 +16,21 @@
 import taichi as ti
 
 
+@taichi_scope
+def expr_init_local_tensor(shape, element_type, elements):
+    ti.current_cfg().make_block_local = False
```

Comment:
Which test(s) does this break?

Reply:
======================================== short test summary info ========================================
FAILED tests/python/test_bls.py::test_gather_3d - RuntimeError: [insert_scratch_pad.cpp:initialize_scr...
FAILED tests/python/test_bls.py::test_gather_1d_trivial - RuntimeError: [insert_scratch_pad.cpp:initia...
FAILED tests/python/test_bls.py::test_scatter_1d_trivial - RuntimeError: [insert_scratch_pad.cpp:initi...
FAILED tests/python/test_bls.py::test_gather_2d_nonsquare - RuntimeError: [insert_scratch_pad.cpp:init...
FAILED tests/python/test_bls.py::test_gather_1d - RuntimeError: [insert_scratch_pad.cpp:initialize_scr...
FAILED tests/python/test_bls.py::test_scatter_2d - RuntimeError: [insert_scratch_pad.cpp:initialize_sc...
FAILED tests/python/test_bls.py::test_gather_2d - RuntimeError: [insert_scratch_pad.cpp:initialize_scr...
FAILED tests/python/test_bls.py::test_scatter_1d - RuntimeError: [insert_scratch_pad.cpp:initialize_sc...
FAILED tests/python/test_bls_assume_in_range.py::test_gathering - RuntimeError: [insert_scratch_pad.cp...

Reply:
Thanks! `make_block_local` has a higher precedence, so let's throw an `InvalidOperationError` here,

Reply:
https://github.com/taichi-dev/taichi/pull/2637/commits/d2af38ee784bbe1624cd41fe476515f1e85c2c19